### ISSUE TITLE
Occasionally doesn't show my full streak stats, just truncated ones

### ISSUE THREAD

**Issue by unixorn** [DenverCoder1/github-readme-streak-stats#288]
**Describe the bug**

Occasionally when I view my profile, the streak stats only go back to Feb 2016. Usually it's correct and shows back to 2014, but every so often I see it truncated.

**To Reproduce**

1. Go to https://github.com/unixorn?tab=overview&from=2014-12-01&to=2014-12-31
2. View the inline streak stats chart

Unfortunately it's intermittent, that will usually show the correct stats, it's only occasionally truncated.

**Expected behavior**

I expect it to always show the same correct stats

**Screenshots**
![incomplete-streak](https://user-images.githubusercontent.com/23920/181811181-ca91142c-8bef-4896-afd6-d399032ea824.svg)

**Desktop (please complete the following information):**

- OS: macOS 12.5 (happened on 12.4 too)
- Browser - I've seen it happen on both Chrome and Firefox

**Additional context**

**Comment by DenverCoder1** [comment]
The stats displayed on the card are directly from the GitHub API.

The reason this may happen is possibly due to timezone-related issues.

It seems from experimenting that the timezone used for the contribution dates is based on the user's last known timezone for the access token provided. I found that 2 of the tokens in the public token pool were last used in a different timezone, but the other four were last used in Denver. To attempt to fix the issue, I've made sure the other two tokens were used in Denver (where I'm located) so it should at least be consistent.

If you self-host the cards using your own personal access token, it would likely resolve the issue as well.

Let me know if this helps or if you continue experiencing the issue.

**Comment by unixorn** [comment]
Will do and thanks for looking into it. Since it's an intermittent thing, it's hard to check.

**Comment by DenverCoder1** [comment]
I mentioned the issue to GitHub Support and they told me "all timestamps are in UTC and this is the expected behavior". If the ability to switch the timezone ever becomes possible, I will be sure to implement it, but for now it seems it is not.

You could always [change the dates](https://codewithhugo.com/change-the-date-of-a-git-commit/) of commits to fill in the gaps if you think it should count when it didn't, but besides that I can't think of anything.

**Comment by unixorn** [comment]
Agreed, if the API is inconsistent you can't do much to cope with it.

**Comment by sumanabhi** [comment]
Hey @DenverCoder1,

I'm also having the same issue, How I can host the card by myself or how we can change it if I want to update the same?
<img width="771" alt="Screenshot 2022-08-03 at 5 04 25 PM" src="https://user-images.githubusercontent.com/20538074/182601330-a4ccdf71-f8a1-44eb-aebe-1e40696f5abf.png">
<img width="525" alt="Screenshot 2022-08-03 at 5 23 36 PM" src="https://user-images.githubusercontent.com/20538074/182601552-80f67f40-d2a0-46ea-99b5-6e09b876ace6.png">

**Comment by DenverCoder1** [comment]
> I'm also having the same issue, How I can host the card by myself or how we can change it if I want to update the same?

The issue again is that the GitHub API says that you have 0 contributions on August 2 even though the graph displays differently.

Based on the response from GitHub Support, I'm not sure that self hosting would make a difference, but you could try it anyway, there are instructions [in the readme](https://github.com/DenverCoder1/github-readme-streak-stats#step-by-step-instructions-for-deploying-to-heroku).

The only way I can think of is to either [change the date](https://codewithhugo.com/change-the-date-of-a-git-commit/) of a past commit or create a new commit dated in the middle of the day on the day that is missing, for example

```bash
git commit --date="2022-08-02 12:00" -m "Test commit" --allow-empty
```

You may consider this to be "cheating" but for now there doesn't seem to be another way to make GitHub count it.

**Comment by TimSchiemann** [comment]
I was having a similar issue and tried resolving it by backdating a commit as you suggested. Now I have a green square on my activity graph but the counter seems to be out of sync. [Here](https://github.com/TimSchiemann) is my profile, and [here](https://github-readme-streak-stats.herokuapp.com?user=TimSchiemann&theme=dark&hide_border=true&date_format=%5BY%20%5DM%20j) is the activity graph that I have embedded. The counter itself works as it should, but the embedded component does not. Any advice?

**Comment by Yohn** [comment]
I just had this timezone thing happen to me.

Its still Aug. 20th, but its showing its the 21st on the pic..
![WrongTimezone](https://user-images.githubusercontent.com/2002591/185772157-41bfba4b-0863-4e4b-aaa1-b05c67eb1e81.svg)

**Comment by eswat2** [comment]
i'm seeing something similar on my stats.  As of just last week, my longest streak was 455, but today it's showing 293 and that appears to be capped by this year (1.1 - 10.20).  My longest streak was actually beyond that...

**Comment by jgphilpott** [comment]
> eswat2

I think all you can do is just wait and it will go back to normal eventually ... seems like the issue is on GitHub's end.

**Comment by DenverCoder1** [comment]
I recently discovered that occasionally the GitHub API was returning blank responses when several requests were sent in a short amount of time, possibly due to rate-limiting.

I have done some testing over the past few days and I believe most of the issues people have been having with missing contributions from past years should be resolved now. In #370, I made it so it will retry requests once if they fail which should greatly reduce the amount of times the streak is missing stats.
